File: aesi-eclipse/bundle/src/main/java/com/virtlink/aesi/structureoutline/Symbol.java
package com.virtlink.aesi.structureoutline;

import com.virtlink.aesi.Span;

public class Symbol implements ISymbol {
    private final String name;
    private final SymbolKind kind;
    private final Span location;

    @Override
    public String getName() { return this.name; }

    @Override
    public SymbolKind getKind() { return this.kind; }

    @Override
    public Span getLocation() { return this.location; }

    public Symbol(String name, SymbolKind kind, Span location) {
        this.name = name;
        this.kind = kind;
        this.location = location;
    }

    @Override
    public String toString() {
        return this.name;
    }
}

File: aesi/src/main/kotlin/com/virtlink/editorservices/symbols/Symbol.kt
package com.virtlink.editorservices.symbols

import com.virtlink.editorservices.Span
import java.net.URI

data class Symbol(
        override val label: String,
        override val resource: URI? = null,
        override val nameRange: Span? = null,
        override val kind: String? = null,
        override val attributes: List<String> = emptyList())
    : ISymbol {

    override fun toString(): String
            = this.label

}